### ISSUE TITLE
Ethernet cable in guide not clear

### ISSUE THREAD

**Issue by ChemicalNRG** [2b-t/velodyne-ros2-docker#1]
The ethernet cable in the guide for anyone without sensor box does display a cross cable.
But must it be cross, or just a normal cable?

**Comment by 2b-t** [comment]
_Hi @ChemicalNRG_ 
Sorry, I wanted to update the cabling section of guide for a while but was quite busy lately.
No, it does not have to be a cross-over cable, it was just the first decent image of an T-568 B wiring I could find. I used [this rhinocables patch cable](https://www.amazon.co.uk/dp/B00DZJNO4M?ref=ppx_yo2ov_dt_b_product_details&th=1), cut it in half and then connected one half and a power connector to the M12 connector. The pin-out of the sensor-side can be found [here](https://pdf.directindustry.com/pdf/velodynelidar/vlp-16-datasheets/182407-676097.html). You have to make the mirrored image of it resulting in the pin-out below:

Pin number | Wire | Signal
-- | -- | --|
![Front view of the cable that you will be making](https://github.com/2b-t/velodyne-ros2-docker/assets/53856473/32098004-407a-47c9-a3ba-04bcced4083d) | | |
1 | Ethernet T-568B green | Ethernet RX-
2 | Ethernet T-568B light-green | Ethernet RX+
3 | Ethernet T-568B orange | Ethernet TX-
4 | Ethernet T-568B light-orange | Ethernet TX+
5 | - | -
6 | - | -
7 | Power cable white | +12V
8 | Power cable black | Ground

Here is a picture of what the final result looks like:
![photo_5807647815965457882_y](https://github.com/2b-t/velodyne-ros2-docker/assets/53856473/ba2291f1-90d6-4a4d-aa3a-9c0d71ba10b0)

Let me know if that answers your question!

**Comment by ChemicalNRG** [comment]
Ok thanks,

And what about gps sync? I think most of the components in the velodyne box are for gps, right?
I dont know if it is a standard sync protocol that velodyne uses, or even if there is such a standard?

Also, is Ground the same as -12v / neutral?

In the manual they say 10.5v - 18v power supply, do you know if the power regulation is done in the interface box or in the lidar? In other words, does it have to be exact 12v or is that just the save middle ground?

[This](https://nl.aliexpress.com/item/4001022938536.html?spm=a2g0o.detail.0.0.4307434bXr0Jyx&gps-id=pcDetailTopMoreOtherSeller&scm=1007.40000.327270.0&scm_id=1007.40000.327270.0&scm-url=1007.40000.327270.0&pvid=4492aa5e-432c-4b6b-b86e-2ac8c9e1748d&_t=gps-id:pcDetailTopMoreOtherSeller,scm-url:1007.40000.327270.0,pvid:4492aa5e-432c-4b6b-b86e-2ac8c9e1748d,tpp_buckets:668%232846%238115%232000&pdp_npi=4%40dis%21EUR%213.58%212.51%21%21%213.70%21%21%40211b5deb16969509038045159e3521%2110000013559492927%21rec%21NL%21161963548%21) is maybe an nicer cable for it.

Also how you feel about protection (reverse and over-voltage protection 5.4.2 of the manual)? Is this the way you would run it in production? Or just in a test environment?

Also it is maybe good to also mention the power supply has to be at least 3A according to 5.4.3 of the manual.

To bad Velodyne is not using the official m12 sensor color coding, that would be a lot easier to follow when using soldered m12 cable. I [found](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.cableforce.cn%2Fen%2Fproduct%2F715.html&psig=AOvVaw0aIuzn6Oqd1j68FNCqWp7F&ust=1697039772709000&source=images&cd=vfe&opi=89978449&ved=0CBEQjRxqFwoTCODF7_jr64EDFQAAAAAdAAAAABAF) [more](https://www.google.com/imgres?imgurl=https%3A%2F%2Fae01.alicdn.com%2Fkf%2FHb78bd101d2c2429b91bc4582a3758513W.jpg_640x640q90.jpg&tbnid=_RHxiLwO1plnDM&vet=12ahUKEwjiouz36-uBAxW7hP0HHcf4Cb8QMygdegUIARCPAQ..i&imgrefurl=https%3A%2F%2Fwww.aliexpress.com%2Fitem%2F1005002459657690.html&docid=uLayi84yXEiZRM&w=640&h=274&q=m12%208%20pin%20connector%20pinout&ved=2ahUKEwjiouz36-uBAxW7hP0HHcf4Cb8QMygdegUIARCPAQ) pinout pictures that are easier to see the difference between male / female for the numbering.

**Comment by 2b-t** [comment]
Thank you for the feedback. I will try to improve the sections that you have pointed out over the course of the following weekend.

If you follow my guide you will **not have GPS time sync**. That would be using pins 5 and 6 which I am not wiring in my guide but in case you need them it should be quite straight forward to wire them in: `GPS_RXD_CNT` should go to pin 5 while `GPS_PULSE_CNT` to pin 6 according to the [schematics in the manual](https://velodynelidar.com/wp-content/uploads/2019/12/63-9243-Rev-E-VLP-16-User-Manual.pdf#page=112) (The way they number the pins and colour code them is sadly a bit confusing ) while you will have to provide the sensor with 5V and ground. The sensor itself has **neither over-voltage nor reverse voltage protection** as stated [here](https://velodynelidar.com/wp-content/uploads/2019/08/63-9259-REV-C-MANUALINTERFACE-BOXHDL-32EVLP-16VLP-32_Web-S.pdf#page=2). From what I understand you have to provide 12V DC as the power regulation is only done by the interface box. So if you need GPS sync or you can't provide a steady 12V DC voltage, either buy the interface box or make your own one starting from the [schematics in the manual](https://velodynelidar.com/wp-content/uploads/2019/12/63-9243-Rev-E-VLP-16-User-Manual.pdf#page=112). Clearly when running this in production I would always add at least a regulator to it but I guess most people using this guide are just like me and are playing around with these sensors in their free time as they are fairly cheap on Ebay at the moment.
That cable looks indeed better than mine!

**Comment by wenchaoliu-93** [comment]
I have looked extensively into the M12 female connector. It seems like there are two reputable brands that carry an array of those products--TE Connectivity and Pheonix Contact. I'd recommend purchasing [this specific product](https://www.digikey.com/en/products/detail/te-connectivity-amp-connectors/T4110002081-000/6679537) from TE for its relatively low cost and for its high build quality.

There are a number of methods to purchase high-quality products at a low cost, but, unfortunately, procurement at AliExpress doesn't seem to be one of them. I dipped my toes, but shipping on AliExpress has been a disappointing experience. The site itself is unorderly, the shipping takes weeks, and the quality of products is unforeseeable at best, unacceptable at worst. Yes, the price was low, but I'd be very hesitant to put my name behind products that were made with those parts.

All that said, I'd be remiss if I didn't vouch for the wiring diagram.
